### ISSUE TITLE
How to use verifier crate?

### ISSUE THREAD

**Issue by matthiasgoergens** [0xPolygonMiden/miden-vm#680]
I am trying to use the stand-alone (verifier)[https://github.com/0xPolygonMiden/miden-vm/tree/next/verifier], but I am having trouble creating all the inputs to the `verify` function.  The `miden-verifier` crate doesn't seem to export eg `ProgramInfo`?  Are we supposed to depend on more crates, instead of just the stand-alone one?

Could we perhaps put a worked example somewhere?  With Rust source and `Cargo.toml` etc?

Thanks for your help!

**Comment by matthiasgoergens** [comment]
My current attempt looks roughly like this (just to get it to compile):

```toml
[package]
name = "verify"
version = "0.1.0"
edition = "2021"

[dependencies]

miden-verifier = { git = "https://github.com/0xPolygonMiden/miden-vm.git", branch = "next" }
miden = { git = "https://github.com/0xPolygonMiden/miden-vm.git", branch = "next" }
```

```rust
use miden::utils::{Deserializable, SliceReader};
use miden::{Digest, Kernel, ProgramInfo, StackInputs, StackOutputs, StarkProof};
use miden_verifier::verify;

fn main() {
    let mut program_hash_slice = SliceReader::new(&[]);
    let digest = Digest::read_from(&mut program_hash_slice)
        .unwrap();

    let kernel: Kernel = Kernel::new(&[]);
    let program_info = ProgramInfo::new(digest, kernel);
    let stack_inputs: StackInputs = StackInputs::new(vec![]);
    let stack_outputs: StackOutputs = StackOutputs::new(vec![], vec![]);
    let proof = StarkProof::from_bytes(&[]).unwrap();
    verify(program_info, stack_inputs, stack_outputs, proof).unwrap();
}
```

**Comment by bobbinth** [comment]
Yeah - that's an omission. `midne-verifier` crate should re-export all relevant structs.

**Comment by matthiasgoergens** [comment]
OK, thanks!  I guess we should make a test to ensure this in future?

**Comment by bobbinth** [comment]
Closed by #683.
